cpp/src/cylon/util/copy_arrray.cpp

```diff
@@ -44,7 +47,7 @@ arrow::Status do_copy_numeric_array(const std::vector<int64_t> &indices,
     if (casted_array->length() <= index) {
```

Comment:
```suggestion
    if (data_array->length() <= index) {
```
We can remove the casted array